rsocket-core/src/main/java/io/rsocket/RSocketServer.java

```diff
@@ -39,15 +42,17 @@
 import static io.rsocket.frame.FrameHeaderFlyweight.FLAGS_M;
 
 /** Server side RSocket. Receives {@link Frame}s from a {@link RSocketClient} */
-class RSocketServer implements RSocket {
+class RSocketServer implements RequestHandler {
 
   private final DuplexConnection connection;
   private final RSocket requestHandler;
+  private final RequestHandler optimizedRequestHandler;
+  private final boolean hasOptimizedRequestHandler;
```

Comment:
No need for an extra boolean property, just checking optimizedRequestHandler != null should be fine